onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1546,24 +1544,30 @@ void SqueezeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const
 
   NodeAttrHelper helper(node);
   vector<int32_t> axes = helper.Get("axes", vector<int32_t>());
-  auto input_dims = shaper[input].size();
+  const auto& input_shape(shaper[input]);
+  auto input_dims = input_shape.size();
   for (auto& axis : axes) {
     if (axis < 0)
       axis += input_dims;
   }
 
-  std::vector<uint32_t> input_indices;
-  input_indices.push_back(operand_indices.at(input));  // input
-
-  if (!axes.empty()) {
-    const auto axes_name = model_builder.GetUniqueName(node.Name() + input + "_axes");
-    Shape axes_dimen = {static_cast<uint32_t>(axes.size())};
-    shaper.AddShape(axes_name, axes_dimen);
-    const OperandType axes_operand_type(Type::TENSOR_INT32, axes_dimen);
-    model_builder.AddOperandFromPersistMemoryBuffer(axes_name, axes.data(), axes_operand_type);
-    input_indices.push_back(operand_indices.at(axes_name));  // axes
+  if (axes.empty()) {  // Squeeze all
+    for (size_t i = 0; i < input_dims; i++) {
+      if (input_shape[i] == 1)
+        axes.push_back(i);
```

Comment:
If we're squeezing I would have expected we want to skip dims where the value is 1.